### PR TITLE
Add instructions for developing with a local version of the action

### PR DESCRIPTION
**Pull request by zimeg**
###  Summary

This PR introduces instructions for developing with a local version of the action to improve the development experience. No longer are pushes and tags to a remote branch necessary.

These changes allow custom configurations to be made to the event payloads to test various cases, and different techniques can be tried with changes to the `local.yml` file.

At the moment only a simple `payload` is provided to mock [Technique 1](https://github.com/slackapi/slack-github-action#technique-1-slack-workflow-builder). I think this is alright to start as more events could be added over time, or experiments can just be performed locally.

Open to any thoughts on this!

### Requirements

* [x] I've read and understood the [Contributing Guidelines](https://github.com/slackapi/slack-github-action/blob/main/.github/contributing.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).

### CODE REVIEW

File: .github/workflows/local/README.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# Local runs
+
+For a simple development experience, a local version of this action can be used in experiments.
+
+**Requirements**:
+
+- An installation of [nektos/act](https://github.com/nektos/act)
```
Comment (seratch) [inline .github/workflows/local/README.md:7]: I didn't know this tool 👍

**Review by seratch — APPROVED** [review]
Looks good to me; it seems the CI builds are failing now although perhaps your changes are unrelated

**Review by WilliamBergamin — APPROVED** [review]
Good additions 💯

**Review by filmaj — APPROVED** [review]
LGTM but yeah we should check the integration test failures

**Comment by zimeg** [comment]
@seratch @WilliamBergamin @filmaj Thank y'all for the reviews! The failing integration tests are strange, but I believe this is caused by this PR coming from a fork.

From the "Actions" tab in settings:

> Anyone with collaborator access to this repository can use these secrets and variables for actions. They are not passed to workflows that are triggered by a pull request from a fork.

It looks like there's [a workaround](https://michaelheap.com/access-secrets-from-forks/) that requires some additions to the workflow file, which might be useful for future contributions! I'll open a new issue to track changes around this, but since this PR doesn't change current behaviors I think it's fine to merge as is.

**Comment by filmaj** [comment]
Hmm that is odd. Usually there is a 'allow the tests to run for this PR' in the case that the PR comes from a fork.

**Comment by zimeg** [comment]
Going to merge this as is since I believe the integration tests are failing due to secrets not being shared with actions run by forks, not from the changes within. Will quickly revert it if failures continue, but continuing research into the failed integration tests in #258 🔍

**Pull request merged**